src/test/java/de/caritas/cob/userservice/api/admin/service/agency/ConsultantAgencyAdminServiceTenantAwareIT.java

```diff
@@ -0,0 +1,209 @@
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import de.caritas.cob.userservice.UserServiceApplication;
+import de.caritas.cob.userservice.agencyadminserivce.generated.web.model.AgencyAdminResponseDTO;
+import de.caritas.cob.userservice.api.admin.service.consultant.create.agencyrelation.ConsultantAgencyRelationCreatorService;
+import de.caritas.cob.userservice.api.exception.httpresponses.BadRequestException;
+import de.caritas.cob.userservice.api.model.AgencyDTO;
+import de.caritas.cob.userservice.api.repository.consultant.Consultant;
+import de.caritas.cob.userservice.api.repository.consultant.ConsultantRepository;
+import de.caritas.cob.userservice.api.repository.consultantagency.ConsultantAgency;
+import de.caritas.cob.userservice.api.repository.consultantagency.ConsultantAgencyRepository;
+import de.caritas.cob.userservice.api.service.agency.AgencyService;
+import de.caritas.cob.userservice.api.tenant.TenantContext;
+import java.util.List;
+import org.jeasy.random.EasyRandom;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.Sql.ExecutionPhase;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = UserServiceApplication.class)
+@TestPropertySource(properties = "spring.profiles.active=testing")
+@AutoConfigureTestDatabase(replace = Replace.ANY)
+@TestPropertySource(properties = "multitenancy.enabled=true")
+@Transactional
+@Sql(value = "/database/setTenantsSpecificData.sql", executionPhase = ExecutionPhase.BEFORE_TEST_METHOD)
+public class ConsultantAgencyAdminServiceTenantAwareIT {
+
+  @Autowired
+  private ConsultantAgencyAdminService consultantAgencyAdminService;
+
+  @Autowired
+  private ConsultantRepository consultantRepository;
+
+  @Autowired
+  private ConsultantAgencyRepository consultantAgencyRepository;
+
+  @MockBean
+  private ConsultantAgencyRelationCreatorService consultantAgencyRelationCreatorService;
+
+  @MockBean
+  private AgencyService agencyService;
+
+  @MockBean
+  private AgencyAdminService agencyAdminService;
+
+  @MockBean
+  private RemoveConsultantFromRocketChatService removeConsultantFromRocketChatService;
+
+  @Before
+  public void beforeTests() {
+    TenantContext.setCurrentTenant(1L);
```

Comment:
From these tests it's hard to guess that we are asserting for a given tenant, let's discuss how we can improve it